crates/core/src/metadata_store.rs

```diff
@@ -24,8 +24,15 @@ use restate_types::storage::{StorageCodec, StorageDecode, StorageEncode, Storage
 use restate_types::{flexbuffers_storage_encode_decode, Version, Versioned};
 use std::future::Future;
 use std::sync::Arc;
+use tonic::Status;
 use tracing::{debug, trace};
 
+#[derive(Debug, thiserror::Error)]
+#[error("{status}")]
+pub struct RemoteError {
+    pub status: Status,
+}
```

Comment:
By putting a `Status` here we basically say that every remote MetadataStore impl must be using Tonic, right? What about etcd, for example?

Reply:
You are absolutely right. This won't work with other stores. I will drop the RemoteError and use only the Internal error. 

> I really think I replied to this yesterday, but for some reason I don't see my comment.